packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1603,12 +1612,37 @@ export class DeltaManager
      * Sorts pending ops and attempts to apply them
      */
     private processPendingOps(reason?: string): void {
-        if (this.handler !== undefined) {
-            const pendingSorted = this.pending.sort((a, b) => a.sequenceNumber - b.sequenceNumber);
-            this.pending = [];
-            // Given that we do not track where these ops came from any more, it's not very
-            // actionably to report gaps in this range.
-            this.enqueueMessages(pendingSorted, `${reason}_pending`, true /* allowGaps */);
+        if (this.closed) {
+            return;
+        }
+
+        assert(this.handler !== undefined, "handler should be installed");
+
+        const pendingSorted = this.pending.sort((a, b) => a.sequenceNumber - b.sequenceNumber);
+        this.pending = [];
+        // Given that we do not track where these ops came from any more, it's not very
+        // actionably to report gaps in this range.
+        this.enqueueMessages(pendingSorted, `${reason}_pending`, true /* allowGaps */);
+
+        // See issue #7312 for more details
```

Comment:
Note: this is the main chance in behavior.
This code get hit a lot (in stress tests) in "read" mode and that condition is describe in comments.
I hit it only once for "write" connection and that's where I believe we have the key to understanding better the underlying problem that is described in 7312. This new code will help "fix" about 50% of cases where we do not see join op for a long time, but telemetry we get in such cases should have a key to why we run into these problems, and hopefully - path to full fix.